### PR TITLE
Save binlog timestampFrom, timestampTo meta when compact

### LINKED ISSUE #25955: [Bug]: With TTL used,we got unusual disk occupation
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.2.3
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

With same profile ,I have start Milvus 2.2.3 for twice. First time I start cluster with TTL , then it comes a unusual disk occupation for 4  million vector occupy 100G disk space on each MinIO (we get four for it ). The second time  without TTL ,50 milion vector only use 15G disk space for on MinIO. Are there  something wrong with using TTL to delete data automatic?
One more thing , we have used Milvus 2.2.8, 2.2.10,2.2.11 with TTL ,they all got unusual disk occupation ,30 miliion vector used almost 1 TB disk space. Maybe there's really a problem with TTL?

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by wayblink**
fix #25955

### CODE REVIEW

File: internal/datacoord/compaction_trigger.go [hunk 1]
```diff
@@ -875,15 +875,22 @@ func (t *compactionTrigger) ShouldDoSingleCompaction(segment *SegmentInfo, isDis
...
 		}
 	}
 
-	if float64(totalExpiredRows)/float64(segment.GetNumOfRows()) >= Params.DataCoordCfg.SingleCompactionRatioThreshold.GetAsFloat() || totalExpiredSize > Params.DataCoordCfg.SingleCompactionExpiredLogMaxSize.GetAsInt64() {
+	if float64(totalExpiredRows)/float64(segment.GetNumOfRows()) >= Params.DataCoordCfg.SingleCompactionRatioThreshold.GetAsFloat() ||
+		totalExpiredSize > Params.DataCoordCfg.SingleCompactionExpiredLogMaxSize.GetAsInt64() {
 		log.Info("total expired entities is too much, trigger compaction", zap.Int64("segmentID", segment.ID),
-			zap.Int("expired rows", totalExpiredRows), zap.Int64("expired log size", totalExpiredSize))
+			zap.Int("expired rows", totalExpiredRows), zap.Int64("expired log size", totalExpiredSize),
+			zap.Bool("segment.CreatedByCompaction", segment.CreatedByCompaction), zap.Int64s("segment.CreatedByCompaction", segment.CompactionFrom))
```
Comment (XuanYang-cn) [inline internal/datacoord/compaction_trigger.go:893]: Seems to have logged two identical keys "segment.CreatedByCompaction"

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]

File: internal/datacoord/compaction_trigger.go (hunk 1 above)
Comment (wayblink) [reply internal/datacoord/compaction_trigger.go:893]: my mistake

Comment (wayblink) [reply internal/datacoord/compaction_trigger.go]: resolved

**Pull request merged**